crates/telio-firewall/src/firewall.rs

```diff
@@ -244,26 +282,40 @@ pub struct StatefullFirewall {
     /// Wheter to still keep track of whitelisted TCP/UDP connections.
     /// Used for connection reset mechanism
     record_whitelisted: bool,
+    /// Local node ip addresses
+    ip_address: RwLock<Vec<StdIpAddr>>,
+    /// Custom IPv4 range to check against
+    custom_ip_range: Option<Ipv4Network>,
 }
 
-#[derive(Debug)]
-struct UdpConnectionInfo {
+#[derive(Debug, PartialEq)]
+struct ConnectionInfo {
     is_remote_initiated: bool,
+    connection_detail: ConnectionDetail,
+}
+
+#[derive(Debug, PartialEq)]
+enum ConnectionDetail {
+    Udp(UdpConnectionDetail),
+    Tcp(TcpConnectionDetail),
+}
+
+#[derive(Debug, PartialEq)]
+struct UdpConnectionDetail {
```

Comment:
At the moment it would be a bit more hassle to make this into a separate PR. I can try to make this into a separate commit to make it easier if you want ?